core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -544,6 +548,7 @@ class KafkaServer(
   private def controlledShutdown(): Unit = {
     val socketTimeoutMs = config.controllerSocketTimeoutMs
 
+    // TODO (KAFKA-14447): Handle controlled shutdown for zkBroker when we have KRaft controller.
```

Comment:
Can we just log a message and `return` from this function early, when we are in this situation? Currently, it looks like we just get stuck and time out, which is not great